### PR TITLE
MM-51701: Show lock output

### PR DESCRIPTION
**Pull request by agnivade**
Example run
```
Starting migration:                        000106_fileinfo_channelid.up.sql---------------------------------------------
LOCKS:
LOCK TYPE: relation, RELATION: fileinfo, MODE: RowExclusiveLock, GRANTED: true
LOCK TYPE: relation, RELATION: posts, MODE: AccessShareLock, GRANTED: true
LOCK TYPE: relation, RELATION: fileinfo, MODE: ShareLock, GRANTED: true
LOCK TYPE: relation, RELATION: fileinfo, MODE: AccessExclusiveLock, GRANTED: true
QUERY:  UPDATE fileinfo SET channelid = posts.channelid FROM posts WHERE fileinfo.channelid IS NULL AND fileinfo.postid = posts.id
EXEC TIME: 0.012882 ms, ROWS AFFECTED: 0
QUERY:  CREATE INDEX IF NOT EXISTS idx_fileinfo_channel_id_create_at ON fileinfo(channelid, createat)
EXEC TIME: 2088.678447 ms, ROWS AFFECTED: 0
QUERY:  SELECT locktype, relation::regclass, mode, granted
                                FROM pg_catalog.pg_locks l
                                LEFT JOIN pg_catalog.pg_database db ON db.oid = l.database
                                JOIN pg_class c ON l.relation=c.oid
                                WHERE (db.datname = current_database() OR db.datname IS NULL) AND c.relkind=$1 AND NOT pid = pg_backend_pid() AND locktype=$2
EXEC TIME: 0.793934 ms, ROWS AFFECTED: 9
QUERY:  ALTER TABLE fileinfo ADD COLUMN IF NOT EXISTS channelid varchar(26)
EXEC TIME: 0.829388 ms, ROWS AFFECTED: 0
```

https://mattermost.atlassian.net/browse/MM-51701

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -73,42 +86,96 @@ func main() {
...
+					fmt.Printf("LOCK TYPE: %s, RELATION: %s, MODE: %s, GRANTED: %s\n", split[0], split[1], split[2], split[3])
+				}
+			}()
+
+			for {
+				select {
+				case <-quit:
+					return
+				case <-ticker.C:
+					rows, err := pool.Query(context.Background(), `SELECT locktype, relation::regclass, mode, granted
```
Comment (isacikgoz) [inline main.go]: Nit: if we are going to add a context timeout sometime, we can use context.TODO instead :p (same for other occurrences)

**Review by isacikgoz — APPROVED** [review]
Looks good, very minor comment :)

File: main.go (hunk 1 above)
Comment (agnivade) [reply main.go]: lol, yes indeed :)

File: main.go [hunk 2]
```diff
@@ -73,42 +86,96 @@ func main() {
...
+				JOIN pg_class c ON l.relation=c.oid
+				WHERE (db.datname = current_database() OR db.datname IS NULL) AND c.relkind='r' AND NOT pid = pg_backend_pid() AND locktype='relation';`)
+					if err != nil {
+						logger.Println(err)
+						return
+					}
+					var lockType, relation, mode string
+					var granted bool
+					pgx.ForEachRow(rows, []any{&lockType, &relation, &mode, &granted}, func() error {
+						lockMap[lockType+"-"+relation+"-"+mode+"-"+fmt.Sprint(granted)] = true
```
Comment (streamer45) [inline main.go]: Personally I'd find a single `fmt.Sprintf` cleaner to generate the key but not a big deal :)

**Review by streamer45 — APPROVED** [review]
Looks good. My only feedback would be that since it's starting to take shape it may help to _functionalize_ some of the logic since there are some cleary defined actions happening in here. Of course I'll leave it to your best judgment :)

**Comment by agnivade** [comment]
> My only feedback would be that since it's starting to take shape it may help to functionalize some of the logic since there are some cleary defined actions happening in here.

Thanks, going to work on that in the next PRs.

File: main.go (hunk 2 above)
Comment (agnivade) [reply main.go]: Done

**Pull request merged**